src/streamlink/plugins/showroom.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import http, validate
+from streamlink.stream import RTMPStream
+
+_url_re = re.compile(r'''^https?://
+        (?:\w*.)?
+        showroom-live.com/
+        (?:
+            (?P<room_title>[\w-]+$)
+            |
+            room/profile\?room_id=(?P<room_id>\d+)$
+        )
+''', re.VERBOSE)
+
+_room_id_re = re.compile(r'"roomId":(?P<room_id>\d+),')
+_room_id_alt_re = re.compile(r'content="showroom:///room\?room_id=(?P<room_id>\d+)"')
+_room_id_lookup_failure_log = 'Failed to find room_id for {0} using {1} regex'
+
+_api_status_url = 'https://www.showroom-live.com/room/is_live?room_id={room_id}'
+_api_data_url = 'https://www.showroom-live.com/room/get_live_data?room_id={room_id}'
+
+_api_data_schema = validate.Schema(
+    {
+        "streaming_url_list_rtmp": validate.all([
+            {
+                "url": validate.text,
+                "stream_name": validate.text,
+                "id": int,
+                "label": validate.text,
+                "is_default": int
+            }
+        ]),
+        "is_live": int,
+        "room": {
+            "room_url_key": validate.text
+        },
+        "telop": validate.any(None, validate.text)
+    }
+)
+_rtmp_quality_lookup = {
+    "オリジナル画質": "original",
+    "original spec": "original",
+    "低画質": "low",
+    "low spec": "low"
+}
+# changes here must also be updated in test_plugin_showroom
+_quality_weights = {
+    "original": 720,
```

Comment:
I'm not sure about the names, it's tricky when they are mixed types. Ideally you want equivalent streams to have the same names so they sort correctly, and the users preferred streaming technology can be picked automatically (see [`--stream-types`](https://streamlink.github.io/cli.html#cmdoption-stream-types).